packages/suite-desktop-core/src/modules/tor.ts

```diff
@@ -12,28 +12,51 @@ import { getFreePort } from '@trezor/node-utils';
 import { validateIpcMessage } from '@trezor/ipc-proxy';
 
 import { TorProcess, TorProcessStatus } from '../libs/processes/TorProcess';
+import { TorExternalProcess } from '../libs/processes/TorExternalProcess';
 import { app, ipcMain } from '../typed-electron';
 
 import type { Dependencies } from './index';
 
 const load = async ({ mainWindowProxy, store, mainThreadEmitter }: Dependencies) => {
     const { logger } = global;
-    const host = '127.0.0.1';
-    const port = await getFreePort();
-    const controlPort = await getFreePort();
-    const torDataDir = path.join(app.getPath('userData'), 'tor');
     const initialSettings = store.getTorSettings();
 
-    store.setTorSettings({ ...initialSettings, host, port });
-
-    const tor = new TorProcess({
-        host,
-        port,
-        controlPort,
-        torDataDir,
-        snowflakeBinaryPath: initialSettings.snowflakeBinaryPath,
+    store.setTorSettings({
+        ...initialSettings,
+        port: await getFreePort(),
+        controlPort: await getFreePort(),
+        torDataDir: path.join(app.getPath('userData'), 'tor'),
     });
 
+    const settings = store.getTorSettings();
+
+    const processes = [
+        {
+            type: 'tor',
+            process: new TorProcess({
+                host: settings.host,
+                port: settings.port,
+                controlPort: settings.controlPort,
+                torDataDir: settings.torDataDir,
+            }),
+        },
+        {
+            type: 'tor-external',
+            process: new TorExternalProcess(),
+        },
+    ];
+
+    const getTarget = () => {
+        const { useExternalTor } = store.getTorSettings();
+        const currentTarget = useExternalTor ? 'tor-external' : 'tor';
+
+        return processes.find(process => process.type === currentTarget)!.process;
+    };
+
```

Comment:
I believe it could be enough to have two `const`s for two different processes here and select one or another based on `useExternalTor`, instead of finding in an array.

Reply:
Sure that works and simpler. https://github.com/trezor/trezor-suite/pull/15642/commits/1d3da5aa4c9aad5d21435018b6104ca50dc27e8b